IBM/carbon-addons-iot-react
# fix(package.json): provide engines field, validate engines in scripts

**Summary**

- I'd like to ensure that we're providing the utmost clarity on which version of Node should be used when developing on the package. We're still running into some very odd snapshot generation issues and testing quirks that appear to be traceable to contributors using Node v12+

**Change List (commits, features, bugs, etc)**

- add the `engines` field to our `package.json`
- add `check-node-version` to validate against `engines` in critical scripts: 
  - `yarn test:*`
  - `yarn start`
  - `yarn build:pre`

**Acceptance Test (how to verify the PR)**

- pull down the branch
- switch your node version to something besides 10.15.3, like `nvm use 12`
- run `yarn` - you should see the default engines error from yarn
```
❯ yarn
yarn install v1.12.3
[1/5] 🔍  Validating package.json...
error carbon-addons-iot-react@0.0.0-development: The engine "node" is incompatible with this module. Expected version "10.15.3". Got "12.16.1"
error Found incompatible module
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```
- run `yarn start` - you should see the error message from `check-node-version`
```
❯ yarn start
yarn run v1.12.3
$ yarn test:engines && cross-env TZ=America/Chicago jest --testPathPattern='.*\.test\.js(x)?' --coverage
$ check-node-version --package
node: 12.16.1
Wanted node version 10.15.3 (10.15.3)
To install node, see https://nodejs.org/download/release/v10.15.3/
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```
